### PR TITLE
Update application layout to new theming

### PR DESCRIPTION
**Pull request by pastelcyborg**
## Done

- Updated `_layout-application` and `_layout-application-panels` to use new theme Sass variables
- Rewrote the `structure` example to better reflect panel and panel child element options
- Removed unnecessary styles from demo CSS
- Added a few missing HTML tags

Fixes [WD-11869](https://warthogs.atlassian.net/browse/WD-11869)

## QA

- Open the [JAAS](https://vanilla-framework-5182.demos.haus/docs/examples/layouts/application/JAAS?theme=light), [default](https://vanilla-framework-5182.demos.haus/docs/examples/layouts/application/default?theme=light). [split](https://vanilla-framework-5182.demos.haus/docs/examples/layouts/application/split?theme=light), and [structure](https://vanilla-framework-5182.demos.haus/docs/examples/layouts/application/structure?theme=light) demos
- Test each theme at each breakpoint size, ensuring each panel/nav element is shown/hidden

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

- [X] PR should have one of the following labels to automatically categorise it in release notes:
  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
- [X] Vanilla version in `package.json` should be updated relative to the [most recent release](https://github.com/canonical/vanilla-framework/releases/latest), following semver convention:
  - if CSS class names are not changed it can be bugfix release (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical/vanilla-framework/wiki/Release-process#pre-release-tasks)
- [X] Any changes to component class names (new patterns, variants, removed or added features) should be listed on the [what's new page](https://github.com/canonical/vanilla-framework/blob/main/releases.yml).

## Screenshots

In lieu of screenshots, would recommend pulling the branch and testing manually; would be numerous screenshots to upload here.

[WD-11869]: https://warthogs.atlassian.net/browse/WD-11869?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by bartaz** [comment]
Looks like this has blown in scope from "update the colours to be themed" to "refactor application layout structure", which makes is a bit harder to review.

Would it be possible to make minimal colour changes separately (just to make sure that any colours used by application layout are themed), and leave the rest of it to be done separately?
Any structure changes need to be backwards compatible, so we have to do them with care.

**Comment by jmuzina** [comment]
Borders on the structure example seem to have a bit too much contrast, unless this is intended:
image

**Comment by bartaz** [comment]
Overall, the structure example is just a structure, it's not about styling at all. So theming doesn't matter to it. We could (and probably should) disable the theming on it completely.

The important bit is if we got rid of any direct colour variables from application/panels code.

**Review by jmuzina — COMMENTED** [review]
The color of the scrollbar for the sidebar (I've been testing [Split](https://vanilla-framework-5182.demos.haus/docs/examples/layouts/application/split?theme=light)) seems to change depending on light/dark theme, despite the sidebar itself always being dark-theme. This makes it particularly hard to see the scrollbar on light-theme.

I'll post a video from my phone in a moment, as Ubuntu's snipping tool is forcing the scrollbar to receive focus / become more visible

https://github.com/canonical/vanilla-framework/assets/46915153/92d2431c-459b-4640-a927-0d4b8f8aa8cb

**Comment by bartaz** [comment]
> The color of the scrollbar for the sidebar (I've been testing [Split](https://vanilla-framework-5182.demos.haus/docs/examples/layouts/application/split?theme=light)) seems to change depending on light/dark theme, despite the sidebar itself always being dark-theme. This makes it particularly hard to see the scrollbar on light-theme.

We haven't been theming (or otherwise altering) native browser scrollbars, so I wouldn't consider it a part of this PR.

If there are any particular issues I'd raise them separately, we would need to have a separate proper look on how theming affects scrollbars across variety of browsers and OSes (and if we can control it in any way).

File: scss/_layouts_application.scss [hunk 1]
```diff
@@ -265,7 +265,7 @@ $application-layout--side-nav-width-expanded: 15rem !default;
 
     @media (min-width: $application-layout--breakpoint-side-nav-collapsed) {
       &.is-pinned {
-        border-left: 1px solid $colors--light-theme--border-low-contrast;
+        border-left: 1px solid $colors--theme--border-low-contrast;
```
Comment (bartaz) [inline scss/_layouts_application.scss:268]:
```suggestion
    border-top: 1px solid $colors--theme--border-default;
```

File: templates/docs/examples/layouts/application/JAAS.html [hunk 2]
```diff
@@ -324,6 +324,7 @@ <h4 class="p-panel__title">Details for 192.168.255.254</h4>
       </div>
     </div>
   </aside>
+</div>
```
Comment (bartaz) [inline templates/docs/examples/layouts/application/JAAS.html:327]: Oh, did we have invalid markup? Good catch!

**Review by bartaz — APPROVED** [review]
Looks good, thanks!

**Pull request merged**